keras_nlp/models/MODEL_CONTRIBUTION_GUIDE.md

```diff
@@ -0,0 +1,204 @@
+with KerasNLP to learn more about the high-level symbols we have.
+
+
+## How is a Backbone model structured in KerasNLP?
+To keep the code simple and readable, we follow
+[Keras' functional style model](https://keras.io/guides/functional_api/) wrapped
+around by a class to implement our models.
+
+A model is typically split into three/four sections. We would recommend you to
+compare this side-by-side with the
+[`keras_nlp.layers.DistilBERTBackbone` source code](https://github.com/keras-team/keras-nlp/blob/v0.4.1/keras_nlp/models/distil_bert/distil_bert_backbone.py#L108-L114)!
+
+
+### Inputs to the model
+
+Generally, the standard inputs to any text model are:
+    - `token_ids`: tokenised inputs (IDs in the form of text).
+    - `padding_mask`: Masks the padding tokens.
+
+
+### Embedding layer(s)
+Standard layers used: `keras.layers.Embedding`,
+`keras_nlp.layers.PositionEmbedding`, `keras_nlp.layers.TokenAndPositionEmbedding`.
+
+
+### Encoder layers
+Standard layers used: `keras_nlp.layers.TransformerEncoder`, `keras_nlp.layers.FNetEncoder`.
+
+
+### Decoder layers (possibly)
+Standard layers used: `keras_nlp.layers.TransformerDecoder`.
+
+
+### Other layers which might be used
+`keras.layers.LayerNorm`, `keras.layers.Dropout`, `keras.layers.Conv1D`, etc.
+
+<br/>
+
+The standard layers provided in Keras and KerasNLP are generally enough for
+99% of the usecases and it is recommended to do a thorough search
+[here](https://keras.io/api/layers/) and [here](https://keras.io/api/keras_nlp/layers/).
+However, sometimes, models have small tweaks/paradigm changes in their architecture.
+This is when things might slightly get complicated.
+
+If the model introduces a paradigm shift, such as using relative attention instead
+of vanilla attention, the contributor will have to implement complete custom layers. A case
+in point is `keras_nlp.models.DebertaV3Backbone` where we had to [implement layers
+from scratch](https://github.com/keras-team/keras-nlp/tree/v0.4.1/keras_nlp/models/distil_bert).
+
+On the other hand, if the model has a small tweak, something simpler can be done.
+For instance, in the Whisper model, the self-attention and cross-attention mechanism
+is exactly the same as vanilla attention, with the exception that the key projection
+layer does not have a bias term. In this case, we can inherit the custom layer
+from one of the standard layers and make minor modifications. See [this PR](https://github.com/keras-team/keras-nlp/pull/801/files#diff-8533ae3a7755c0dbe95ccbb71f85c677297f687bf3884fadefc64f1d0fdce51aR22) for
+more details.
+
+## Steps
+
+To make a complete model contribution, three PRs have to be opened:
+
+1) "Add `DistilBertBackbone`" (with a Colab link demonstrating correct model implementation and weight conversion)
+2) "Add `DistilBertTokenizer` and `DistilBertPreprocessor`"
+3) "Add `DistilBertBackbone` Presets"
+
+
+### Open an issue/Claim an open issue
```

Comment:
Claim -> find

Given how many people we have looking for issues, we should discourage a "stake your claim" vibe :)